kotlin/export-tiles/src/main/java/com/example/exporttiles/MainActivity.kt

```diff
@@ -0,0 +1,263 @@
+                if (status == Job.Status.SUCCEEDED) {
+                  showMapPreview(result)
+                  downloadArea?.isVisible = false
+
+                } else {
+                  ("Job did not succeed: " + error.additionalMessage).also {
+                    Toast.makeText(this@MainActivity, it, Toast.LENGTH_LONG).show()
+                    Log.e(TAG, error.additionalMessage)
+                  }
+                }
+              }
+            }
+        } catch (e: InterruptedException) {
+          Toast.makeText(this, "TileCacheParameters interrupted: " + e.message, Toast.LENGTH_LONG)
+            .show()
+          Log.e(TAG, "TileCacheParameters interrupted: " + e.message)
+        } catch (e: ExecutionException) {
+          Toast.makeText(this, "Error generating parameters: " + e.message, Toast.LENGTH_LONG)
+            .show()
+          Log.e(TAG, "Error generating parameters: " + e.message)
+        }
+      }
+    }
+
+    // get correct view order set up on start
+    clearPreview(mapView)
+  }
+
+  /**
+   * Show tile cache preview window including containing the exported tiles.
+   *
+   * @param result takes the TileCache from the ExportTileCacheJob
+   */
+  private fun showMapPreview(result: TileCache) {
+    val newTiledLayer = ArcGISTiledLayer(result)
+    val previewMap = ArcGISMap(Basemap(newTiledLayer))
+
+    // set up the preview map view
+    previewMapView.apply {
+      map = previewMap
+      setViewpoint(mapView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE))
+    }
+    // control UI visibility
+    previewMapView.getChildAt(0).visibility = View.VISIBLE
+    show(closeButton, dimBackground, preview_text_view, previewMapView)
+    exportTilesButton.visibility = View.GONE
+
+    // // required for some Android devices running older OS (combats Z-ordering bug in Android API)
+    mapPreviewLayout.bringToFront()
+  }
+
+  /**
+   * Clear the preview window.
+   */
+  fun clearPreview(view: View) {
+
+    previewMapView.getChildAt(0).visibility = View.INVISIBLE
+    hide(closeButton, dimBackground, preview_text_view, previewMapView)
+    // control UI visibility
+    show(exportTilesButton, mapView)
+    downloadArea?.isVisible = true
+
+    // required for some Android devices running older OS (combats Z-ordering bug in Android API)
+    mapView.bringToFront()
+  }
+
+  /**
+   * Create a progress dialog box for tracking the export tile cache job.
+   *
+   * @param exportTileCacheJob the export tile cache job progress to be tracked
+   * @return an AlertDialog set with the dialog layout view
+   */
+  private fun createProgressDialog(exportTileCacheJob: ExportTileCacheJob): AlertDialog {
+    val builder = AlertDialog.Builder(this@MainActivity).apply {
+      setTitle("Exporting tiles...")
+      // provide a cancel button on the dialog
+      setNeutralButton("Cancel") { _, _ ->
+        exportTileCacheJob.cancel()
+      }
+      setView(R.layout.dialog_layout)
+    }
```

Comment:
```suggestion
    val builder = AlertDialog.Builder(this@MainActivity).apply {
      setTitle("Exporting tiles...")
      // provide a cancel button on the dialog
      setNeutralButton("Cancel") { _, _ ->
        exportTileCacheJob.cancel()
      }
      setCancelable(false)
      setView(R.layout.dialog_layout)
    }
```

Reply:
Adding `setCancelable(false)` will make it so the user tapping outside the dialog or hitting back won't stop the download, but the "cancel" button will still work :)